server/routes/__tests__/money.spec.js

```diff
@@ -0,0 +1,327 @@
+const request = require('supertest');
+const cheerio = require('cheerio');
+const createAxiosRequestError = require('axios/lib/core/createError');
+
+const { createMoneyRouter } = require('../money');
+const {
+  PrisonerInformationService,
+} = require('../../services/prisonerInformation');
+const { PrisonApiRepository } = require('../../repositories/prisonApi');
+const { User } = require('../../auth/user');
+
+const {
+  setupBasicApp,
+  consoleLogError,
+} = require('../../../test/test-helpers');
+
+describe('GET /money/transactions', () => {
+  let app;
+  const client = { get: jest.fn() };
+
+  const prisonApiRepository = new PrisonApiRepository({ client });
+  const prisonerInformationService = new PrisonerInformationService({
+    prisonApiRepository,
+  });
+  const moneyRouter = createMoneyRouter({
+    hubContentService: () => {},
+    offenderService: () => {},
+    prisonerInformationService,
+  });
+
+  const testUser = new User({
+    prisonerId: 'PRISONER_ID',
+    firstName: 'Test',
+    surname: 'User',
+    bookingId: 'BOOKING_ID',
+  });
+
+  const transactionApiResponse = [
+    {
+      entryDate: '2021-02-23',
+      transactionType: 'TELE',
+      entryDescription: 'Television',
+      currency: 'GBP',
+      penceAmount: 50,
+      accountType: 'SPND',
+      postingType: 'DR',
+      agencyId: 'TST',
+      currentBalance: -443,
+    },
+  ];
+
+  const agencyApiResponse = {
+    agencyId: 'TST',
+    description: 'Test (HMP)',
+    longDescription: 'HMP TEST',
+  };
+
+  const balancesApiResponse = {
+    spends: 123.45,
+    cash: 456.78,
+    savings: 890.12,
+    currency: 'GBP',
+  };
+
+  const fourOhFour = createAxiosRequestError('🤷‍♂️', null, 404);
+  const fiveOhThree = createAxiosRequestError('💥', null, 503);
+
+  const setMockUser = (req, res, next) => {
+    req.user = testUser;
+    next();
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    app = setupBasicApp();
+  });
+
+  it('should prompt to sign in when the user is signed out', async () => {
+    app.use('/money', moneyRouter);
+
+    await request(app)
+      .get('/money/transactions')
+      .expect(200)
+      .then(response => {
+        const $ = cheerio.load(response.text);
+        expect($('.govuk-body').text()).toMatch(/sign in/im);
```

Comment:
nice!, we do really rubbish text presence check and have kept meaning to look for something like this

Reply:
😄